src/azul/chalice.py

```diff
@@ -467,6 +495,33 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
+    def threshold(self, *, errors: int, throttles: int):
+        def wrapper(f):
+            f.errors_threshold = errors
+            f.throttles_threshold = throttles
+            return f
+
+        return wrapper
+
+    def metric_thresholds(self) -> list[MetricThreshold]:
+        default_threshold = 0
+        thresholds = []
+        lambda_name, _ = config.unqualified_resource_name(self.app_name)
+        for metric in LambdaMetric:
+            # The api_handler lambda functions (indexer & service) aren't included
+            # in the app_module's handler_map, so we account for those first.
+            thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                              metric=metric,
+                                              value=default_threshold))
+            for handler in self.handler_map.values():
+                if isinstance(handler, chalice.app.EventSourceHandler):
+                    threshold = getattr(handler, f'{metric.name}_threshold', default_threshold)
+                    thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                                      handler_name=getattr(handler, 'name'),
```

Comment:
```suggestion
                                                      handler_name=handler.name,
```

Reply:
I've added this change as a `drop!` commit since this change works (results in the same generated content for `cloudwatch.tf.json`), however PyCharm flags this as a warning
```
Unresolved attribute reference 'name' for class 'EventSourceHandler'
```